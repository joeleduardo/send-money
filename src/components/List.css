ul {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

li {
  position: relative;
  display: flex;
  flex-direction: column;
  border-bottom: 1px solid #cccccc;
  padding: .5rem;
  margin-bottom: .5rem;
}

li span {
  position: absolute;
  right: 0;
}

.name {
  margin-bottom: .3rem;
}

.name:first-letter {
  text-transform: uppercase;
}

.email {
  color: #999999;
  font-size: .8rem;
}