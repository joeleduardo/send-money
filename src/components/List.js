import React from 'react';
import PropTypes from 'prop-types';
import './List.css';

const List = (props) => {
  const { items } = props;

  const numberWithCommas = (amount) => {
    return amount.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
  }

  return(
    <ul>
      {items.map((item) => (
        <li key={item.id}>
          <div className='name'>{item.name}</div>
          <div className='email'>{item.email}</div>
          <span>Â£ {numberWithCommas(item.amount)}</span>
        </li>
      ))}
    </ul>
  )
}

List.propTypes = {
  items: PropTypes.array.isRequired,
};

List.defaultProps = {
  items: []
};

export default List;