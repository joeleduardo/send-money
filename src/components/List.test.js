import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

const items = [
  {
    id: 'rFj2aAOIsILzHmP29VHo',
    name: 'Joel Ortiz',
    email: 'joeleduardo@hotmail.com',
    amount: '1000.00'
  }
];

test('List component', () => {
  const cList = shallow(<List items={items} />);
  expect(cList).toMatchSnapshot();
});
