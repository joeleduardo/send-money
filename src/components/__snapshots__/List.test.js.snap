// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    items={
        Array [
            Object {
              "amount": "1000.00",
              "email": "joeleduardo@hotmail.com",
              "id": "rFj2aAOIsILzHmP29VHo",
              "name": "Joel Ortiz",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <div
                    className="name"
          >
                    Joel Ortiz
          </div>
          <div
                    className="email"
          >
                    joeleduardo@hotmail.com
          </div>
          <span>
                    £ 
                    1,000.00
          </span>
</li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "rFj2aAOIsILzHmP29VHo",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="name"
>
              Joel Ortiz
</div>,
            <div
              className="email"
>
              joeleduardo@hotmail.com
</div>,
            <span>
              £ 
              1,000.00
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Joel Ortiz",
              "className": "name",
            },
            "ref": null,
            "rendered": "Joel Ortiz",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "joeleduardo@hotmail.com",
              "className": "email",
            },
            "ref": null,
            "rendered": "joeleduardo@hotmail.com",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "£ ",
                "1,000.00",
              ],
            },
            "ref": null,
            "rendered": Array [
              "£ ",
              "1,000.00",
            ],
            "type": "span",
          },
        ],
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <div
                        className="name"
            >
                        Joel Ortiz
            </div>
            <div
                        className="email"
            >
                        joeleduardo@hotmail.com
            </div>
            <span>
                        £ 
                        1,000.00
            </span>
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "rFj2aAOIsILzHmP29VHo",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="name"
>
                Joel Ortiz
</div>,
              <div
                className="email"
>
                joeleduardo@hotmail.com
</div>,
              <span>
                £ 
                1,000.00
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Joel Ortiz",
                "className": "name",
              },
              "ref": null,
              "rendered": "Joel Ortiz",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "joeleduardo@hotmail.com",
                "className": "email",
              },
              "ref": null,
              "rendered": "joeleduardo@hotmail.com",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "£ ",
                  "1,000.00",
                ],
              },
              "ref": null,
              "rendered": Array [
                "£ ",
                "1,000.00",
              ],
              "type": "span",
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
